File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/DateConverter.java
package org.ccci.gto.android.common.room.converter;

import java.util.Date;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

public class DateConverter {
    @Nullable
    @TypeConverter
    public static Date toDate(@Nullable final Long timestamp) {
        return timestamp != null ? new Date(timestamp) : null;
    }

    @Nullable
    @TypeConverter
    public static Long toLong(@Nullable final Date date) {
        return date != null ? date.getTime() : null;
    }
}

File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/DateConverter.kt
package org.ccci.gto.android.common.room.converter

import androidx.room.TypeConverter
import java.util.Date

@Deprecated("Since v3.4.0, use converter from the gto-support-androidx-room module instead")
object DateConverter {
    @JvmStatic
    @TypeConverter
    fun toDate(timestamp: Long?) = timestamp?.let { Date(it) }

    @JvmStatic
    @TypeConverter
    fun toLong(date: Date?) = date?.time
}
